src/common/low_precision_transformations/src/recurrent_cell.cpp

```diff
@@ -91,8 +62,132 @@ RecurrentCellTransformation::RecurrentCellTransformation(const Params& params) :
     this->register_matcher(m, callback);
 }
 
+namespace {
+
+std::shared_ptr<ov::opset1::FakeQuantize> find_fake_quantize_upper(const std::shared_ptr<Node>& parent) {
+    if (is_type<ov::opset1::FakeQuantize>(parent)) {
+        return as_type_ptr<ov::opset1::FakeQuantize>(parent);
+    }
+
+    if (!NetworkHelper::isPrecisionPreserved(parent)) {
+        return nullptr;
+    }
+
+    return find_fake_quantize_upper(parent->get_input_node_shared_ptr(0));
+}
+
+template <class Operation>
+std::string name() {
+    return Operation::get_type_info_static().name;
+}
+
+bool isSupportedForPerChannelQuantization(const std::shared_ptr<Node>& node) {
+    static const std::unordered_set<std::string> precisionPreservedOps = {
+        { name<opset1::DepthToSpace>() },
+        { name<opset1::Interpolate>() },
+        { name<opset1::MaxPool>() },
+        { name<opset1::ReduceMax>() },
+        { name<opset1::ReduceMin>() },
+        { name<opset1::Relu>() },
+        { name<opset2::BatchToSpace>() },
+        { name<opset1::Broadcast>() },
+        { name<opset3::Broadcast>() },
+        { name<opset1::Pad>() },
+        { name<opset12::Pad>() },
+        { name<opset1::Reshape>() },
+        { name<opset1::Squeeze>() },
+        { name<opset2::SpaceToBatch>() },
+        { name<opset1::Split>() },
+        { name<opset1::StridedSlice>() },
+        { name<opset1::ShuffleChannels>() },
+        { name<opset1::Transpose>() },
+        { name<opset1::Unsqueeze>() },
+        { name<opset1::VariadicSplit>() }
+    };
+
+    return precisionPreservedOps.find(node->get_type_name()) != precisionPreservedOps.end();
+}
+
+std::vector<std::pair<size_t, element::Type>> get_quantized_inputs(std::shared_ptr<ov::Node> lstm) {
+    return is_type<ov::opset5::LSTMSequence>(lstm) ?
+        std::vector<std::pair<size_t, element::Type>>{ {0, element::u8}, { 1, element::u8 }, { 4, element::undefined }, { 5, element::undefined } } :
```

Comment:
We always regulate the desired precisions on inputs with `PrecisionsRestriction` which are set by plugin. In CPU, we actually have u8u8 restrictions:
```
            PrecisionsRestriction::create<ov::opset5::LSTMSequence>({
                    {{0, 1}, {ov::element::u8}}
                }),
            PrecisionsRestriction::create<ov::opset6::GRUSequence>({
                    {{0, 1}, {ov::element::u8}}
                }),
```

Why do we need this additional logic with precisions limitations here? If there is no chance to avoid it, could you please add an explanatory comment in the code on what `undefined` precision means?

Reply:
LPT uses operation input precision restrictions to identify output precision for `FakeQuantize` dequantization transformation. But it doesn't mean, that we can ignore filter or input validation in each transformation pattern matching, because in Constant branch (for example, a branch with single `Constant` operation) LPT can not provide require precision.

Reply:
This is a valid point, but it seems like we have such checks only in this transformation. Does it mean that we must perform the checks in each transformation (or at least in the transformations which work with layers with weights)?